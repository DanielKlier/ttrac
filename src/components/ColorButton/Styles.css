.color-button .color-tile {
    width: 24px;
    height: 24px;
}
.color-button .btn {
    padding: 4px;
}

.color-button .btn[disabled] {
    cursor: default;
}

.color-button.selected .btn {
    border: 1px solid #337ab7;
    box-shadow: 0 0 1px #337ab7;
}
