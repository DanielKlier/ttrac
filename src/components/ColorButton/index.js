import React from 'react';
import PropTypes from 'prop-types';
import {noop} from 'lodash';
import {Button} from 'react-bootstrap';
import './Styles.css';

function ColorButton(props) {

    let wrapperClass = 'color-button';

    if (props.isSelected) {
        wrapperClass += ' selected';
    }

    return (
        <div className={wrapperClass}>
            <Button onClick={() => props.onClick(props.color)} disabled={props.disabled}>
                <div className="color-tile" style={{backgroundColor: props.color}}/>
            </Button>
        </div>
    );
}

ColorButton.propTypes = {
    color: PropTypes.string.isRequired,
    isSelected: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
};

ColorButton.defaultProps = {
    color: '#555555',
    isSelected: false,
    onClick: noop,
    disabled: false
};

export default ColorButton;
