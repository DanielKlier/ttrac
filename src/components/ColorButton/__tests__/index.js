import React from 'react';
import {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ColorButton from '../index';

const color = '#ffaaff';

test('renders a color button normally', () => {
    const component = mount(<ColorButton color={color}/>);

    expect(toJSON(component)).toMatchSnapshot();
});

test('it calls the click handler', () => {
    const handleClick = jest.fn();

    const component = mount(<ColorButton color={color} onClick={handleClick}/>);

    component.find('Button').simulate('click');

    expect(handleClick).toBeCalledWith(color);
});

test('cannot be clicked when disabled', () => {
    const handleClick = jest.fn();

    const component = mount(<ColorButton color={color} onClick={handleClick} disabled={true}/>);

    component.find('Button').simulate('click');

    expect(handleClick).toHaveBeenCalledTimes(0);
});

test('renders its selected state', () => {
    const component = mount(<ColorButton color={color} isSelected={true}/>);

    expect(toJSON(component)).toMatchSnapshot();
});
