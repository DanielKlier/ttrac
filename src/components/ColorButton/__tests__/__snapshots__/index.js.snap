// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a color button normally 1`] = `
<ColorButton
  color="#ffaaff"
  disabled={false}
  isSelected={false}
  onClick={[Function]}
>
  <div
    className="color-button"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <div
          className="color-tile"
          style={
            Object {
              "backgroundColor": "#ffaaff",
            }
          }
        />
      </button>
    </Button>
  </div>
</ColorButton>
`;

exports[`renders its selected state 1`] = `
<ColorButton
  color="#ffaaff"
  disabled={false}
  isSelected={true}
  onClick={[Function]}
>
  <div
    className="color-button selected"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <div
          className="color-tile"
          style={
            Object {
              "backgroundColor": "#ffaaff",
            }
          }
        />
      </button>
    </Button>
  </div>
</ColorButton>
`;
